relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -684,35 +684,16 @@ mod tests {
             DecayingFunction::Linear {
                 decayed_sample_rate: 0.2,
             },
-            None,
-            None,
-        );
-
-        assert!(
-            prepare_and_get_sampling_rule(1.0, EventType::Transaction, &project_state, now)
-                .unwrap()
-                .is_none()
-        );
-    }
-
-    #[test]
-    fn test_event_decaying_rule_with_no_time_range_and_constant_function() {
-        let now = Utc::now();
-        let project_state = state_with_decaying_rule(
-            Some(0.7),
-            RuleType::Transaction,
-            SamplingMode::Total,
-            DecayingFunction::Constant,
-            None,
-            None,
+            Some(now - DateDuration::days(1)),
+            Some(now + DateDuration::days(1)),
         );
 
         assert_eq!(
             prepare_and_get_sampling_rule(1.0, EventType::Transaction, &project_state, now)
                 .unwrap()
                 .unwrap()
                 .sample_rate,
-            0.7
+            0.44999999999999996
```

Comment:
This looks flaky, should we rather do something like `(expected - result).abs() < eps` instead? See https://rust-lang.github.io/rust-clippy/master/#float_cmp

Reply:
Very good call! I totally overlooked the precision, especially considering that it can differ from architecture to architecture.